packages/diagram/lib/diagram.dart

```diff
@@ -0,0 +1,442 @@
+      _builder = builder;
+      _binding.updateDiagram(builder);
+    }
+  }
+
+  /// The output directory used when a [File] generated by the
+  /// [frameFilenameGenerator], or the `outputFile` argument to
+  /// [drawDiagramToFile] is a relative path. The written file will be relative
+  /// to [outputDirectory]. If the filename is absolute, then [outputDirectory]
+  /// is ignored.
+  ///
+  /// Defaults to [Directory.current].
+  ///
+  /// On a device, this will need to be set to the result of
+  /// [getApplicationDocumentsDirectory] (from the path_provider package) or a
+  /// subdirectory of that directory to have sufficient permissions to write
+  /// files.
+  Directory outputDirectory;
+
+  /// The generator for filenames when calling drawAnimatedDiagramToFile.
+  ///
+  /// If the returned filenames are relative paths, they will be relative to
+  /// [outputDirectory].
+  AnimationFrameFilenameGenerator get frameFilenameGenerator => _frameFilenameGenerator;
+  AnimationFrameFilenameGenerator _frameFilenameGenerator = _basicFrameFilenameGenerator;
+  set frameFilenameGenerator(AnimationFrameFilenameGenerator frameFilenameGenerator) {
+    _frameFilenameGenerator = frameFilenameGenerator ?? _basicFrameFilenameGenerator;
+  }
+
+  _DiagramFlutterBinding get _binding => _DiagramFlutterBinding.instance();
+
+  /// Advances the animation clock by the given duration.
+  ///
+  /// The [increment] must be greater than, or equal to, [Duration.zero].
+  void advanceTime(Duration increment) => _binding.pump(duration: increment);
+
+  /// Returns an [image.Image] representing the current diagram.
+  ///
+  /// The returned image will have the dimensions of the widget generated by
+  /// [builder] in logical coordinates, multiplied by the [pixelRatio].
+  ///
+  /// Time will be advanced by [duration] before taking the snapshot.
+  Future<ui.Image> drawDiagramToImage({Duration duration: Duration.zero}) async {
+    advanceTime(duration);
+    return _binding.takeSnapshot();
+  }
+
+  /// Draws the widget returned by [builder] and writes it to [outputFile].
+  ///
+  /// The file will be written in whatever format the suffix on the file
+  /// indicates. Valid suffixes are ".jpg", ".jpeg", ".png", ".gif", and ".tga".
+  Future<File> drawDiagramToFile(
+    File outputFile, {
+    ui.ImageByteFormat format: ui.ImageByteFormat.png,
+  }) async {
+    assert(outputFile != null);
+    if (!outputFile.isAbsolute && outputDirectory != null) {
+      // If output path is relative, make it relative to the output directory.
+      outputFile = new File(path.join(outputDirectory.absolute.path, outputFile.path));
+    }
+    ui.Image captured = await drawDiagramToImage();
+    ByteData encoded = await captured.toByteData(format: format);
+    List<int> bytes = encoded.buffer.asUint8List().toList();
+    print('Writing ${bytes.length} bytes, ${captured.width}x${captured.height} '
+        '${_byteFormatToString(format)}, to ${outputFile.absolute.path}.');
```

Comment:
to avoid the problem of the trailing `.` looking like it's part of the filename, you could remove it and instead put a colon after `to`.